docs/api-fields.md

```diff
@@ -0,0 +1,203 @@
+# AWS ProviderSpec and ProviderStatus documentation
+
+## AWS ProviderSpec
+
```

Comment:
i think it would be nice to show a full example here of the provider spec, then the following sections could be seen as a reference back to the example.